src/lib/raft/raft.h

```diff
@@ -261,6 +273,15 @@ raft_vote_count(const struct raft *raft)
 	return raft->votes[raft->self].count;
 }
 
+/** Time since the latest communication with the leader. */
+static inline double
+raft_leader_idle(const struct raft *raft)
+{
+	if (raft->is_enabled && raft->state != RAFT_STATE_LEADER)
```

Comment:
Why do you skip it if raft is disabled? Disabled nodes still keep track of raft state. If that is only for `box.info.election`, I would suggest moving the being enabled check there if it is still necessary somewhy.

Reply:
Well, `raft_process_heartbeat` doesn't update anything when raft is disabled, so leader_idle() would mean nothing on a disabled node.
I think it'd be better to return 0 and not some random number.
Or better let's do this:
```diff
 static inline double
 raft_leader_idle(const struct raft *raft)
 {
-       if (raft->is_enabled && raft->state != RAFT_STATE_LEADER) {
+       assert(raft->is_enabled);
+       if (raft->state != RAFT_STATE_LEADER) {
                return raft_ev_monotonic_now(raft_loop()) -
                       raft->leader_last_seen;
        }

```
and in next commit:
```diff
diff --git a/src/box/lua/info.c b/src/box/lua/info.c
index 0336d2b2f..7df4c436e 100644
--- a/src/box/lua/info.c
+++ b/src/box/lua/info.c
@@ -623,8 +623,10 @@ lbox_info_election(struct lua_State *L)
        lua_setfield(L, -2, "vote");
        lua_pushinteger(L, raft->leader);
        lua_setfield(L, -2, "leader");
-       lua_pushnumber(L, raft_leader_idle(raft));
-       lua_setfield(L, -2, "leader_idle");
+       if (raft_is_enabled(raft)) {
+               lua_pushnumber(L, raft_leader_idle(raft));
+               lua_setfield(L, -2, "leader_idle");
+       }
        return 1;
 }
 

```